sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/HealthcareFhirResourceCreatedEventData.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.messaging.eventgrid.systemevents;
+
+import com.azure.core.annotation.Immutable;
+
+/** Schema of the Data property of an EventGridEvent for a Microsoft.HealthcareApis.FhirResourceCreated event. */
+@Immutable
+public final class HealthcareFhirResourceCreatedEventData extends HealthcareFhirResourceEventBaseProperties {
+    /** {@inheritDoc} */
+    @Override
+    public HealthcareFhirResourceCreatedEventData setResourceType(HealthcareFhirResourceType resourceType) {
+        super.setResourceType(resourceType);
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public HealthcareFhirResourceCreatedEventData setResourceFhirAccount(String resourceFhirAccount) {
+        super.setResourceFhirAccount(resourceFhirAccount);
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public HealthcareFhirResourceCreatedEventData setResourceFhirId(String resourceFhirId) {
```

Comment:
Why do we have setter methods on a type that is annotated with `@Immutable`?

Reply:
change to annotation Fluent